core/datastore/src/main/proto/com/google/samples/apps/nowinandroid/data/user_preferences.proto

```diff
@@ -43,4 +43,9 @@ message UserPreferences {
 
     ThemeBrandProto theme_brand = 16;
     DarkThemeConfigProto dark_theme_config = 17;
+
+    // Note that proto3 only allows a default value of `false` for boolean types. This means that
+    // whilst the name `should_show_onboarding` would be preferable here we are forced to choose a
+    // name which works with this default value constraint.
+    bool has_dismissed_onboarding = 18;
```

Comment:
Would `shouldHideOnboarding` work as well?

Reply:
Yeah, it's still not ideal (`hide` is harder to understand than `show`), but better than `has_dismissed` which, to quote adamp, is an "event masquerading as state". 

Code updated. Thanks for the review 👍